nibabel/tests/test_optpkg.py

```diff
@@ -0,0 +1,63 @@
+""" Testing optpkg module
+"""
+
+import types
+import sys
+from distutils.version import LooseVersion
+
+from nose import SkipTest
+from nose.tools import (assert_true, assert_false, assert_raises,
+                        assert_equal)
+
+
+from nibabel.optpkg import optional_package
+from nibabel.tripwire import TripWire, TripWireError
+
+
+def assert_good(pkg_name, min_version=None):
+    pkg, have_pkg, setup = optional_package(pkg_name, min_version=min_version)
+    assert_true(have_pkg)
+    assert_equal(sys.modules[pkg_name], pkg)
+    assert_equal(setup(), None)
+
+
+def assert_bad(pkg_name, min_version=None):
+    pkg, have_pkg, setup = optional_package(pkg_name, min_version=min_version)
+    assert_false(have_pkg)
+    assert_true(isinstance(pkg, TripWire))
+    assert_raises(TripWireError, getattr, pkg, 'a_method')
+    assert_raises(SkipTest, setup)
+
+
+def test_basic():
+    # We always have os
+    assert_good('os')
+    # Subpackage
+    assert_good('os.path')
+    # We never have package _not_a_package
+    assert_bad('_not_a_package')
+
+
+def test_versions():
+    fake_name = '_a_fake_package'
+    fake_pkg = types.ModuleType(fake_name)
+    assert_false('fake_pkg' in sys.modules)
+    # Not inserted yet
+    assert_bad(fake_name)
+    try:
+        sys.modules[fake_name] = fake_pkg
+        # No __version__ yet
+        assert_good(fake_name)  # With no version check
+        assert_bad(fake_name, '1.0')
+        # We can make an arbitrary callable to check version
+        assert_good(fake_name, lambda pkg : True)
```

Comment:
These aren't hitting the Travis tests, but are still PEP8 violations. Just out of curiosity, is this becoming a convention in other projects?